src/redux/posts/posts.actions.js

```diff
@@ -12,14 +12,16 @@ import {
   ADD_POST,
 } from './posts.types';
 
+import censorBadWords from '../../services/censorBadWords'
+
 // Get posts
 export const getPosts = () => async (dispatch) => {
   try {
     const res = await axios.get(config.BASE_URL + '/api/posts');
 
     dispatch({
       type: GET_POSTS,
-      payload: res.data.data,
+      payload: res.data.data.map(({title, body, ...rest})=>({...censorBadWords({title, body}), ...rest})),
```

Comment:
Over here there's no loop, but to remain consistency and to avoid insert and delete operation we can just convert the text while rendering